### ISSUE TITLE
Cannot install packages from source with RTools 4.2 and renv (Windows)

### ISSUE THREAD

**Issue by martinmodrak** [rstudio/renv#1041]
Hi,
On my Windows 10 PC, I just upgraded to R 4.2.1 and RTools 4.2. When `renv` is activated, I cannot install packages from source:

```r
 install.packages("base64enc", type = "source")
```

```
* installing *source* package 'base64enc' ...
** package 'base64enc' successfully unpacked and MD5 sums checked
** using staged installation
** libs
/mingw64/bin/gcc  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c base64.c -o base64.o
/bin/sh: line 1: /mingw64/bin/gcc: No such file or directory
make: *** [C:/PROGRA~1/R/R-42~1.1/etc/x64/Makeconf:253: base64.o] Error 127
ERROR: compilation failed for package 'base64enc'
```

However, in a session without active `renv`, the installation proceeds without problems:

```
* installing *source* package 'base64enc' ...
** package 'base64enc' successfully unpacked and MD5 sums checked
** using staged installation
** libs
gcc  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c base64.c -o base64.o
gcc  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c uriencode.c -o uriencode.o
gcc  -I"C:/PROGRA~1/R/R-42~1.1/include" -DNDEBUG     -I"C:/rtools42/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c utf8.c -o utf8.o
gcc -shared -s -static-libgcc -o base64enc.dll tmp.def base64.o uriencode.o utf8.o -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib/x64 -LC:/rtools42/x86_64-w64-mingw32.static.posix/lib -LC:/PROGRA~1/R/R-42~1.1/bin/x64 -lR
installing to C:/Users/modrak_m/AppData/Local/R/win-library/4.2/00LOCK-base64enc/00new/base64enc/libs/x64
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (base64enc)
```

The difference appears to be that when `renv` is active, the installation tries to use an explicit path for `gcc` (`/mingw64/bin/gcc`) which however doesn't exist, while without `renv`, no path is provided for `gcc` which is then found via the `PATH` variable.

I however have no idea how/where `renv` could introduce this change into the compilation process, thanks for any hints.

Potentially related to #638 but looks quite different to me.

**Comment by kevinushey** [comment]
Can you confirm what version of `renv` you're using? This sounds like an issue that may be resolved in the current CRAN release (0.15.5).

**Comment by martinmodrak** [comment]
You are right, I had 0.15.4 and upgrading to 0.15.5 resolves the problem! Thanks
